awsshell/utils.py

```diff
@@ -22,6 +26,25 @@ def build_config_file_path(file_name):
     return os.path.join(os.path.expanduser('~'), '.aws', 'shell', file_name)
 
 
+@contextlib.contextmanager
+def temporary_file(mode):
+    """Cross platform temporary file creation.
+
+    This is an alternative to ``tempfile.NamedTemporaryFile`` that
+    also works on windows and avoids the "file being used by
+    another process" error.
+    """
+    tempdir = tempfile.mkdtemp()
```

Comment:
Instead of creating a temp directory and then deleting it later with `shutil.rmtree(tempdir)`, does getting the temp directory with `tempfile.gettempdir()` simplify this process?